el/src/main/java/jakarta/enterprise/inject/spi/el/ELSupport.java

```diff
@@ -0,0 +1,31 @@
+package jakarta.enterprise.inject.spi.el;
+
+import jakarta.el.ELResolver;
+import jakarta.el.ExpressionFactory;
+import jakarta.enterprise.context.Dependent;
+
+/**
+ * Allows integrators to obtain Unified EL objects that are integrated with the CDI container
+ * as described in the CDI specification.
+ */
+public interface ELSupport {
```

Comment:
@Ladicek My thinking was we should introduce an interface that extends BM to keep it similar to what we had before.

Also, are beans with EL names globally available?
I am asking because BM can be (in case of Weld at least) caller-sensitive so the set of resolveable beans you can reach by performing `@Inject BeanManager` differs based on caller. This allows Weld to mimic EE visibility rules and can be seen in more complex deployments such as EARs running on WFLY.
If beans with EL names are also subject to this, that's an extra reason to keep this interface as an extension to BM.

Reply:
> I don't really see why this needs to extend BeanManager. There's nothing preventing classes that implement BeanManager from also implementing ELSupport, that should be pretty much it I guess?

Yes, I imagine impls would ship something like EL-aware BM impl.
Which is why it might as well be on an interface extending BM.
That would allow users to inject that new interface and gain access to both, EL functionality as well as BM functionality.

> Perhaps the spec chapter on EL should mention this visibility concern, that's a good point.

I don't think these rules are specified even for BM, not in much detail anyway. And if I recall correctly, different impls (OWB) varied in how they did it.

Reply:
Actually if we create `interface ELAwareBeanManager extends BeanManager { ... }`, then we can specify that in Jakarta EE, the `BeanManager` is always an `ELAwareBeanManager`, which allows everyone to obtain it using `CDI.current().getBeanManager()` (and casting), which is probably desirable. I'll change that and add some spec text for this.

Reply:
Done.